pkg/reconciler/ingress/ingress.go

```diff
@@ -163,7 +156,11 @@ func (r *Reconciler) reconcileIngress(ctx context.Context, ing *v1alpha1.Ingress
 		if err := r.reconcileWildcardGateways(ctx, desiredWildcardGateways, ing); err != nil {
 			return err
 		}
-		wildcardGatewayNames = resources.GetQualifiedGatewayNames(desiredWildcardGateways)
+
+		// VirtualService will be attached to both global Gateways and the Knative generated Gateways.
+		// We still want to attach to the global Gateways to respect any global Gateway configuration.
```

Comment:
How does this work though?
By global Gateway, I am assuming you mean a `*` Gateway?
In this case, the more specific Gateway will be picked and other ignored, no?

Reply:
The more specific Gateway (or Knative generated Gateway) only has the TLS configuration.
In this PR, I plan to just simply move the TLS configuration from the global Gateway to the more specific Gateway. So for the TLS part, there is no conflict when attaching VirtualService to both global and the more specific Gateway.

And by attaching VirtualService to the global Gateway, I want to retain the non-TLS behavior for the backward compatibility reason.